src/sentry/integrations/slack/tasks.py

```diff
@@ -120,3 +125,32 @@ def find_channel_id_for_rule(project, actions, uuid, rule_id=None, **kwargs):
         return
     # if we never find the channel name we failed :(
     redis_rule_status.set_value("failed")
+
+
+@instrumented_task(
+    name="sentry.integrations.slack.search_channel_id_metric_alerts", queue="integrations"
+)
+def find_channel_id_for_alert_rule(organization_id, uuid, **kwargs):
+    redis_rule_status = RedisRuleStatus(uuid)
+    try:
+        organization = Organization.objects.get(id=organization_id)
+    except Organization.DoesNotExist:
+        redis_rule_status.set_value("failed")
+        return
+
+    kwargs["use_async_lookup"] = True
+    serializer = AlertRuleSerializer(
+        context={"organization": organization, "access": SystemAccess()}, data=kwargs,
```

Comment:
A bit worried about using `SystemAccess` here. I can imagine a situation where we have a slack action and an email action which is sending to a team. The email action requires checking permissions, but maybe when we initially tried to save the AlertRule we checked slack first, it raised `ChannelLookupTimeoutError` and we never checked permissions on the email action.

Thinking about ways to handle this:
 - Not sure if there's a good way to pass permissions through to another task, probably not.
 - Another option could be to create the AlertRule and not include the slack action, and do just the slack part async.
 - Simplest might be that when we detect that we need to look up slack async we continue validating the rest of the alert rule, and if it passes then we raise `ChannelLookupTimeoutError`. Then it's safe to use `SystemAccess` here.

Reply:
@wedamija Well, we should not even call the task unless the serializer passed the first time. If you look at the endpoint, the error `ChannelLookupTimeoutError` error is raised during `save` which isn't called unless we pass the serializer.

Reply:
@wedamija Good callout. I have refactored the serializers so that we only raise the `ChannelLookupTimeoutError` if there are no there validation errors.